use docsy page edit/issue setup

kbhawkey opened the pull request:
- Use theme's edit page and create issue links on page.
- Clean up "edit" button on page. Related to issue #22172.
- Investigating style of feedback buttons.
- Created override for theme `toc.html` partial. Related to issue #23132.

kbhawkey commented:
Page previews:
https://deploy-preview-23227--kubernetes-io-master-staging.netlify.app/docs/reference/command-line-tools-reference/kube-apiserver/

https://deploy-preview-23227--kubernetes-io-master-staging.netlify.app/docs/concepts/architecture/control-plane-node-communication/

https://deploy-preview-23227--kubernetes-io-master-staging.netlify.app/docs/home/

May want to move the edit/log an issue for the blog pages?
https://deploy-preview-23227--kubernetes-io-master-staging.netlify.app/blog/

sftim commented on config.toml:
```diff
@@ -127,6 +127,9 @@ nextUrl = "https://kubernetes-io-vnext-staging.netlify.com/"
 githubWebsiteRaw = "raw.githubusercontent.com/kubernetes/website"
 
+# GitHub repository links for editing a page and opening issues.
+github_repo = "https://github.com/kubernetes/website"
```
This is similar to `githubWebsiteRepo`; I'd either merge these or use consistent capitalization (eg: change `githubWebsiteRepo` to `github_repo_short_name`)

Maybe something to log as an issue for some follow-up work?

kbhawkey replied:
Yes, I will remove the other variables if there are no dependencies.

kbhawkey replied:
This variable is used by the theme.

sftim replied:
Future tidying: we could use the new `github_repo` parameter in `layouts/partials/git-info.html`, to generate the links to page history.

kbhawkey replied:
Instead, I added in the theme's git-info partial, "page-meta-lastmod.html" and updated the
date format in config.toml.

sftim reviewed (commented):
/lgtm
It makes sense to adopt a Docsy approach.

kbhawkey commented:
Note: In cleaning up, I removed the git change info at the bottom of each page. Adding back.

kbhawkey commented:
Note: log an issue to create localized strings.
Replace "main_edit_this_page" with theme variable: "post_edit_this" 
Replace "main_github_create_an_issue" with theme variable: "post_create_issue"

sftim commented:
/lgtm

There's room to iterate but that's IMO no reason not to merge.

kbhawkey commented:
This is a WIP.

kbhawkey commented:
/area web-development

kbhawkey commented on layouts/partials/feedback.html:
```diff
@@ -2,8 +2,8 @@
     <p class="feedback--prompt">{{ T "feedback_question" }} </p>
-    <button class="btn btn-primary mb-4 feedback--yes">{{ T "feedback_yes" }}</button>
-    <button class="btn btn-primary mb-4 feedback--no">{{ T "feedback_no" }}</button>
+    <button class="btn mb-4 feedback--yes">{{ T "feedback_yes" }}</button>
```
Not sure if this is the preferred way to handle customizing the buttons.
Could override in _variables_project.scss instead of adding to _custom.scss.

celestehorgan replied:
I'd prefer if we used Bootstrap's classes rather than customizing CSS personally! ✨

kbhawkey replied:
Yeah, I agree. Though, the current styling is not great.

kbhawkey replied:
I left the feedback buttons as oval. Could set the `border-radius`.
For now, leaving as is.

sftim commented:
@kbhawkey you can /work-in-progress if that's appropriate

Not sure if that was your intention.

kbhawkey commented:
> 
> 
> @kbhawkey you can /work-in-progress if that's appropriate
> 
> Not sure if that was your intention.

I think the changes are ready for review. Some things to check:
- The feedback buttons are not sized equally. I could continue to work on this, leave as is, or revert the changes. With the theme updates, the buttons in the site use the defaults instead of the custom colors and shapes.
- I added a toc.html to override the theme's toc.html. This change adds a toc for the component tool pages where there are few headings but large tables/page size. I've logged an issue with the theme. Think it might be useful to create a variable to represent the character length threshold for displaying the toc.
- Previously, the blog pages displayed a toc. I removed the toc from the blog pages. There is no edit/create issue on blog pages.
- I swapped out the current git info, edit page, create issue for the theme's page-meta-links.html partial.
Could investigate using the existing "pencil" icon in the page-meta-links.html partial instead of the defaults.

sftim commented:
Seems a clear improvement with no problem regressions.
/lgtm

celestehorgan commented:
This is awesome @kbhawkey ✨ 

One change I'd like to request:

<img width="221" alt="Screen Shot 2020-09-03 at 10 31 41 AM" src="https://user-images.githubusercontent.com/17999517/92147984-babab480-edd0-11ea-967d-f2e645c512a0.png">

Could we go into `i18n.toml` and change **Create issue** to **Create issue?**. Documentation is sort of implied, and the extra word causes the text to wrap on smaller resolutions like my laptop.

kbhawkey commented:
> 
> 
> This is awesome @kbhawkey ✨
> 
> One change I'd like to request:
> 
> <img alt="Screen Shot 2020-09-03 at 10 31 41 AM" width="221" src="https://user-images.githubusercontent.com/17999517/92147984-babab480-edd0-11ea-967d-f2e645c512a0.png">
> 
> Could we go into `i18n.toml` and change **Create documentation issue** to **Create issue**? Documentation is sort of implied, and the extra word causes the text to wrap on smaller resolutions like my laptop.

I think I will leave this for another time. I do see the wrapped text on my laptop screen too. Thanks!

zacharysarah commented:
@kbhawkey Thanks so much for this! The icons in the right bar are ✨ 

Thanks for factoring for i18n! 👍 

/lgtm
/approve
